native/KotlinIOS/KotlinIOS/ViewController.swift

```diff
@@ -1,20 +1,72 @@
 import UIKit
 import SharedCode
 
-class ViewController: UIViewController {
-
-    @IBOutlet private var label: UILabel!
-
+class ViewController: UIViewController, UIPickerViewDataSource, UIPickerViewDelegate {
+    @IBOutlet weak var originSpinner: UIPickerView!
+    @IBOutlet weak var destinationSpinner: UIPickerView!
+    @IBOutlet weak var submitButton: UIButton!
+    
+    private var stations: [String] = [String]()
+    
     private let presenter: ApplicationContractPresenter = ApplicationPresenter()
     
     override func viewDidLoad() {
         super.viewDidLoad()
         presenter.onViewTaken(view: self)
+        
+        originSpinner.delegate = self
+        originSpinner.dataSource = self
+        
+        destinationSpinner.delegate = self
+        destinationSpinner.dataSource = self
     }
+    
+    func numberOfComponents(in pickerView: UIPickerView) -> Int {
+        return 1
+    }
+    
+    func pickerView(_ pickerView: UIPickerView, numberOfRowsInComponent component: Int) -> Int {
+        return stations.capacity
+    }
+    
+    func pickerView(_ pickerView: UIPickerView, titleForRow row: Int, forComponent component: Int) -> String? {
+        return stations[row]
+    }
+    
+
 }
 
 extension ViewController: ApplicationContractView {
-    func setLabel(text: String) {
-        label.text = text
+    func getDestinationStation() -> String {
+        let row = destinationSpinner.selectedRow(inComponent: 0)
+        return stations[row]
+    }
+    
+    func getOriginStation() -> String {
+        let row = originSpinner.selectedRow(inComponent: 0)
+        return stations[row]
+    }
+    
+    func openUrl(url: String) {
+        UIApplication.shared.open(URL(string: url)!)
+        return
+    }
+    
+    func setStationSubmitButtonHandler() {
+        submitButton.addTarget(self, action: #selector(buttonPress(_:)), for: UIControl.Event.touchDown)
+    }
+    
+    @objc func buttonPress(_ sender: UIButton){
+        // TODO: better method to do this?
+        presenter.onStationSubmitButtonPressed()
+    }
```

Comment:
I think you can replace the above two functions by using the drag and drop method to drag 'touchUpInside' to your file, like you do when creating a reference to your component:
<img width="259" alt="Screenshot 2022-06-28 at 16 55 47" src="https://user-images.githubusercontent.com/52420057/176225558-6fd5b079-b3ec-4824-b684-58e095473888.png">

Reply:
Then I'd expect something like this is you view controller:
```
@IBAction func stationSubmitButtonTapped(_ sender: Any) {
        presenter.onStationSubmitButtonPressed()
    }
```